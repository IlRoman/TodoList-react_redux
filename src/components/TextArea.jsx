import React from 'react';
import PropTypes from 'prop-types';

export class TextArea extends React.Component {
    state = {
        content: ''
    }

    handleChange = (element) => {
        const { name, value } = element.target
        this.setState({
            [name]: value
        })
    }

    render() {
        return (
            <div className="create-task">
                <input
                    name="content"
                    className="create-task__input"
                    type="text"
                    value={this.state.value}
                    onChange={this.handleChange}
                />
                <button
                    className="btn create-task-btn"
                    onClick={() => this.props.onCreateTask(this.state.content)}
                >Create</button>
            </div>
        );
    }
}

TextArea.propTypes = {
    onCreateTask: PropTypes.func,
}