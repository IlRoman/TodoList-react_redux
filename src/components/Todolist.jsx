import React from 'react';
import { TextArea } from './TextArea';
import { TasksList } from './TasksList';
import { connect } from 'react-redux';
import * as actions from '../redux/actions'
import { tasksListSelector } from '../redux/selectors'
import PropTypes from 'prop-types';

class TodoList extends React.Component {
    componentDidMount() {
        this.props.getTasksList();
    }

    render() {
        const sortedList = this.props.tasks
            .slice()
            .sort((a, b) => a.done - b.done);

        return (
            <>
                <h1 className="title">Todo list</h1>
                <main className="todo-list">
                    <TextArea onCreateTask={this.props.onCreateTask} />
                    <ul className="list">
                        {sortedList.map(task =>
                            <TasksList
                                key={task.id}
                                id={task.id}
                                text={task.text}
                                done={task.done}
                                onDelete={this.props.onDeleteTask}
                                onChange={this.props.onUpdateTask}
                            />
                        )}
                    </ul>
                </main>
            </>
        )
    }
}

TodoList.propTypes = {
    text: PropTypes.string,
    done: PropTypes.bool,
    id: PropTypes.string,
}

const mapState = state => {
    return {
        tasks: tasksListSelector(state)
    }
}

const mapDispatch = {
    getTasksList: actions.getTasksList,
    onUpdateTask: actions.onUpdateTask,
    onDeleteTask: actions.onDeleteTask,
    onCreateTask: actions.onCreateTask,
}

export default connect(mapState, mapDispatch)(TodoList)