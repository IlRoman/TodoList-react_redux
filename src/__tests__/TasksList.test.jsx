import React from 'react';
import { shallow } from 'enzyme';
import { TasksList } from '../components/TasksList'

describe('<TasksList />', () => {
    it('should display done task', () => {
        const props = {
            text: 'task 1',
            done: true,
        }
        const wrappedComponent = shallow(<TasksList {...props} />)
        expect(wrappedComponent).toMatchSnapshot()
    })

    it('should display undone task', () => {
        const props = {
            text: 'task 1',
            done: false,
        }
        const wrappedComponent = shallow(<TasksList {...props} />)
        expect(wrappedComponent).toMatchSnapshot()
    })

    it('should update task on checkbox checked', () => {
        const props = {
            id: '1',
            onChange: jest.fn(),
            text: 'task 1',
            done: false,
        }
        const wrappedComponent = shallow(<TasksList {...props} />)
        wrappedComponent.find('.list-item__checkbox').simulate('change')
        expect(props.onChange).toBeCalledWith('1')
    })

    it('should delete task', () => {
        const props = {
            id: '1',
            onChange: jest.fn(),
            text: 'task 1',
            done: false,
            onDelete: jest.fn()
        }
        const wrappedComponent = shallow(<TasksList {...props} />)
        wrappedComponent.find('.list-item__delete-btn').simulate('click')
        expect(props.onDelete).toBeCalledWith('1')
    })
})