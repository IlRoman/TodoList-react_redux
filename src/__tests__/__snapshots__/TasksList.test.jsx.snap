// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TasksList /> should display done task 1`] = `
<li
  className="list-item list-item_done"
>
  <input
    className="list-item__checkbox"
    defaultChecked={true}
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="list-item__text"
  >
    task 1
  </span>
  <button
    className="list-item__delete-btn"
    onClick={[Function]}
  />
</li>
`;

exports[`<TasksList /> should display undone task 1`] = `
<li
  className="list-item"
>
  <input
    className="list-item__checkbox"
    defaultChecked={false}
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="list-item__text"
  >
    task 1
  </span>
  <button
    className="list-item__delete-btn"
    onClick={[Function]}
  />
</li>
`;

exports[`<TasksList /> should update task on checkbox checked 1`] = `[MockFunction]`;
